my-app/src/Utilities/ValidationUtilities.tsx

```diff
@@ -123,6 +125,42 @@ export function isValidMove(
                .map(point => getPieceAtLocation(point))
                   .some(piece => piece !== null);      
    }
+
+   /*  
+      castling is only possible if neither the king nor the rook has moved
+      there must not be any pieces between the king and the rook
+      the king may not be in check
+      the square the king goes to and any intervening squares may not be under attack
+      */
+   function pieceIsCastlingAndCannotCastle(moveLocation: Coordinate, piece: Piece, board: Array<Array<(Piece | null)>>) : boolean {
+      if (piece.pieceName !== Pieces.KING) return false;
+      if (!(piece as King).isCastlingMove(moveLocation)) return false;
+
+      let correspondingCastle: Piece | null = getCorrespondingCastle(moveLocation, board);
+
+      if (!correspondingCastle || correspondingCastle.pieceName !== Pieces.ROOK || correspondingCastle.hasMoved) {
```

Comment:
I think it may be good to check that the rook is the same color as the castling king, but that case should be ruled out by the move putting the king in check

Reply:
Yeah, I can update it just in case